core/app/models/spree/variant.rb

```diff
@@ -152,7 +152,7 @@ def option_value(opt_name)
     end
 
     def price_in(currency)
-      prices.select{ |price| price.currency == currency }.first || Spree::Price.new(variant_id: self.id, currency: currency)
+      prices.find{ |price| price.currency == currency } || Spree::Price.new(variant_id: self.id, currency: currency)
```

Comment:
Prefer `detect` over `find`.<br>Redundant `self` detected.<br>Space missing to the left of {.